### PR TITLE
ROC-4754 Event Creation for CCJ

### PR DESCRIPTION
**Pull request by karthikkannankainos**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-4754

### Change description ###
Generating Default CCJ

### CODE REVIEW

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java [hunk 1]
```diff
@@ -44,29 +48,34 @@ public CCDCountyCourtJudgment to(CountyCourtJudgment countyCourtJudgment) {
             builder.ccjType(CCDCountyCourtJudgmentType.valueOf(ccjType.name()))
         );
 
+        Optional.ofNullable(claim.getCountyCourtJudgmentRequestedAt()).ifPresent(builder::requestedDate);
```
Comment (dharmendrak) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: no need to check optional if present as we are not processing on date, as well as by default it is already defaulted to null. so no harm in setting the date directly in builder

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java [hunk 2]
```diff
@@ -17,13 +17,17 @@
 import static uk.gov.hmcts.cmc.ccd.domain.CCDPaymentOption.valueOf;
 
 @Component
-public class CountyCourtJudgmentMapper implements Mapper<CCDCountyCourtJudgment, CountyCourtJudgment> {
+public class CountyCourtJudgmentMapper {
 
-    @Override
-    public CCDCountyCourtJudgment to(CountyCourtJudgment countyCourtJudgment) {
+    public CCDCountyCourtJudgment to(Claim claim) {
+
```
Comment (dharmendrak) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: in all mapper we are checking for        
        ```Objects.requireNonNull(claim, "claim must not be null");```

File: src/main/java/uk/gov/hmcts/cmc/claimstore/events/ccd/CCDCaseHandler.java [hunk 3]
```diff
@@ -121,7 +121,7 @@ public void requestMoreTimeForResponse(CCDMoreTimeRequestedEvent event) {
         }
     }
```
Comment (dharmendrak) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/events/ccd/CCDCaseHandler.java]: no point changing this in transactional as calling service method is not transactional

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapper.java [hunk 4]
```diff
@@ -21,15 +22,18 @@
     private final TheirDetailsMapper theirDetailsMapper;
     private final ResponseMapper responseMapper;
     private final ClaimantResponseMapper claimantResponseMapper;
+    private final CountyCourtJudgmentMapper ccjMapper;
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapper.java]: Please rename this to `countyCourtJudgmentMapper`

File: src/main/java/uk/gov/hmcts/cmc/claimstore/events/ccd/CCDCaseHandler.java (hunk 3 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/events/ccd/CCDCaseHandler.java]: It is transactional in `CountyCourtJudgmentService`. So let it be `@TransactionalEventListener`

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java [hunk 5]
```diff
@@ -185,4 +188,21 @@ public void mapTheirDetailsFromCCDClaimWithResponse() {
             finalClaim.isMoreTimeRequested(), ccdDefendant.getResponseMoreTimeNeededOption().toBoolean());
     }
 
+    @Test
+    public void mapCCJFromToCCDDefendant() {
+        //Given
+        TheirDetails theirDetails = SampleTheirDetails.builder().organisationDetails();
+        Claim claimWithCCJ = SampleClaim.getClaimFullDefenceStatesPaidWithAcceptation();
```
Comment (sabahirfan) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java]: Here as discussed, don't populate the defendant response. It should be claim with default CCJ, and not defendant response.

**Comment by sabahirfan** [comment]
Please remove these deprecated classes:
`uk.gov.hmcts.cmc.ccd.deprecated.mapper.ccj.RepaymentPlanMapper`
`uk.gov.hmcts.cmc.ccd.deprecated.domain.CCDRepaymentPlan`
`uk.gov.hmcts.cmc.ccd.deprecated.assertion.RepaymentPlanAssert`
`uk.gov.hmcts.cmc.ccd.deprecated.mapper.ccj.RepaymentPlanMapperTest`

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java (hunk 5 above)
Comment (karthikkannankainos) [reply ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java]: done

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java [hunk 6]
```diff
@@ -185,4 +188,21 @@ public void mapTheirDetailsFromCCDClaimWithResponse() {
             finalClaim.isMoreTimeRequested(), ccdDefendant.getResponseMoreTimeNeededOption().toBoolean());
     }
 
+    @Test
+    public void mapCCJFromToCCDDefendant() {
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/DefendantMapperTest.java]: changed it to mapDefaultCCJToCCDDefendant

File: src/main/java/uk/gov/hmcts/cmc/claimstore/events/ccd/CCDCaseHandler.java (hunk 3 above)
Comment (karthikkannankainos) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/events/ccd/CCDCaseHandler.java]: left it as @TransactionalEventListener. Tested in local and was able to see CCJ is posted to CCD

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java (hunk 1 above)
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: Defensive programming is always good. But I have changed as asked.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java (hunk 2 above)
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: we are returning null if we cant get details. So I am returning null if either claim or claim.getCCJ is null.

Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ccj/CountyCourtJudgmentMapper.java]: Claim to asserted null should be part of claimmapper.

**Comment by karthikkannankainos** [comment]
@sabahirfan will remove uk.gov.hmcts.cmc.ccd.deprecated.mapper.ccj.RepaymentPlanMapper
uk.gov.hmcts.cmc.ccd.deprecated.domain.CCDRepaymentPlan
uk.gov.hmcts.cmc.ccd.deprecated.assertion.RepaymentPlanAssert
uk.gov.hmcts.cmc.ccd.deprecated.mapper.ccj.RepaymentPlanMapperTest as part of following PR,

**Pull request merged**